cg/apps/housekeeper/hk.py

```diff
@@ -662,3 +664,40 @@ def reset_retrieved_archive_data(self, files_to_reset: list[File]):
             file.archive.retrieval_task_id = None
             file.archive.retrieved_at = None
         self.commit()
+
+    @staticmethod
+    def get_files_with_tags(files: list[File], tags: list[set[str]]) -> list[File]:
+        """Return files containing specified tags."""
+        filtered_files: list[File] = []
+        for file in files:
+            file_tags: set[str] = {tag.name for tag in file.tags}
+            if any(tag.issubset(file_tags) for tag in tags):
+                filtered_files.append(file)
+        return filtered_files
+
+    @staticmethod
+    def get_files_without_excluded_tags(
+        files: list[File], excluded_tags: list[set[str]]
```

Comment:
```suggestion
    def get_files_without_tags(
        files: list[File], tags: list[set[str]]
```
I get confused by the "excluded" here 😅

Reply:
I think using `excluded_tags` makes it clearer and easier to distinguish from `get_files_with_tags`. 

And defining the tags as list of sets offers more flexibility for excluding files, like those matching two specific tags

Reply:
We'll hopefully merge the sample and case tags soon, and after that, I believe we could further simplify these functions